.github/workflows/github_pr.yml

```
@@ -61,12 +60,40 @@ jobs:
         sudo apt-get install libsodium-dev
         sudo apt-get install g++-9
 
+    - name: '[Windows] Install dependencies & setup environment'
+      if: runner.os == 'Windows'
+      shell: powershell
+      run: |
+        $url = "https://download.libsodium.org/libsodium/releases/libsodium-1.0.18-msvc.zip"
+        $sha256hash = "C1D48D85C9361E350931FFE5067559CD7405A697C655D26955FB568D1084A5F4"
+        Write-Host ('Downloading {0} ...' -f $url)
+        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        # See https://github.com/PowerShell/PowerShell/issues/2138
+        $ProgressPreference = 'SilentlyContinue'
+        Invoke-WebRequest -Uri $url -OutFile 'libsodium.zip'
+        if ((Get-FileHash libsodium.zip -Algorithm "SHA256").Hash -ne $sha256hash) {
+          exit 1
+        }
+        Expand-Archive libsodium.zip -DestinationPath d:\a\agora\agora\source\scpp\lib\
```

```suggestion
        Expand-Archive libsodium.zip -DestinationPath ${{ github.workspace }}\agora\source\scpp\lib\
```

SSame for the other instances of `d:\a\agora\`

---

Applied suggest.

---

Hum it looks like the replacement is for `d:\a\agora\agora\`, not `d:\a\agora\`